sw/host/opentitanlib/src/app/mod.rs

```diff
@@ -653,10 +678,28 @@ impl TransportWrapper {
     /// Returns a [`GpioPin`] implementation.
     pub fn gpio_pin(&self, name: &str) -> Result<Rc<dyn GpioPin>> {
         let resolved_pin_name = map_name(&self.pin_map, name);
-        if let Some(pin) = self.artificial_pin_map.get(&resolved_pin_name) {
-            return Ok(pin.clone());
+        let mut pin_instance_map = self.pin_instance_map.borrow_mut();
+        // Find if we already have a GpioPinWrapper around the requested instance.  If
+        // not, create one.
+        if let Some(instance) = pin_instance_map.get(&resolved_pin_name) {
+            Ok(Rc::clone(instance) as Rc<dyn GpioPin>)
+        } else {
+            let instance = if let Some(pin) = self.artificial_pin_map.get(&resolved_pin_name) {
+                pin.clone()
+            } else {
+                self.transport.gpio_pin(resolved_pin_name.as_str())?
+            };
+            let mut invert = false;
+            if let Some(conf) = self.pin_conf_map.get(&resolved_pin_name) {
+                if let Some(inv) = conf.invert {
+                    invert = inv;
+                }
+            }
```

Comment:
```suggestion
           let invert = self.pin_conf_map.get(&resolved_pin_name).unwrap_or(false).unwrap_or(false);
```

Reply:
The suggestion wouldn't type check. If should be 
```rust
let invert = self.pin_conf_map.get(&resolved_pin_name).and_then(|conf| conf.invert).unwrap_or(false);
```

Reply:
Done